modules/tinymce/src/core/main/ts/fmt/ExpandRange.ts

```diff
@@ -38,9 +38,9 @@ const findParentContentEditable = (dom: DOMUtils, node: Node) => {
 
 const walkText = (start: boolean, node: Text, offset: number, predicate: (chr: string) => boolean) => {
   const str = node.data;
-  for (let i = offset; start ? i >= 0 : i < str.length; start ? i-- : i++) {
-    if (predicate(str.charAt(i))) {
-      return start ? i + 1 : i;
+  for (let i = offset; start ? i > 0 : i < str.length; start ? i-- : i++) {
+    if (predicate(str.charAt(start ? i - 1 : i))) {
+      return i;
```

Comment:
I wonder if this would be better as two separate loops this is crazy complex now.

- [ ] Consider splitting up the loops to two loops

```suggestion
  if (start) {
   for (let i = offset; i > 0; i--) {
       if (predicate(str.charAt(i - 1))) {
         return i; // Should this be i - 1?
       }
    }
  } else {
   for (let i = offset; i < str.length; i++) {
       if (predicate(str.charAt(i))) {
         return i;
       }
   }
 }
```

Reply:
I did think about it, but I wasn't sure it was any more readable, or needed.

Reply:
It also gets a bit tricky to read when separate, but I get you. I'll change it.